docs/root/intro/arch_overview/outlier.rst

```diff
@@ -10,6 +10,39 @@ such as consecutive failures, temporal success rate, temporal latency, etc. Outl
 form of *passive* health checking. Envoy also supports :ref:`active health checking
 <arch_overview_health_checking>`. *Passive* and *active* health checking can be enabled together or
 independently, and form the basis for an overall upstream health checking solution.
+Outlier detection is part of :ref:`cluster configuration <envoy_api_msg_cluster.OutlierDetection>` 
+and it needs filters to report errors, timeouts, resets. Currently the following filters support
+outlier detection: :ref:`http router <config_http_filters_router>`, 
+:ref:`tcp proxy <config_network_filters_tcp_proxy>`  and :ref:`redis proxy <config_network_filters_redis_proxy>`.
+
+Type of detected errors depends on filter type. :ref:`http router <config_http_filters_router>` filter for example
+detects locally originated errors (timeouts, resets - errors related to connection to upstream host) and because it 
+also understands HTTP protocol it reports 
+errors returned by HTTP server. In such scenario, even when connection to upstream HTTP server is successful,  
+transaction with the server may fail. Such errors are called here transaction errors or externally originated errors.
+On the contrary, :ref:`tcp proxy <config_network_filters_tcp_proxy>` filter does not understand any protocol above
+TCP layer and reports only locally originated (connection) errors.
+
+In default configuration (:ref:`outlier_detection.split_external_local_origin_errors<envoy_api_field_cluster.OutlierDetection.split_external_local_origin_errors>` is *false*) 
+locally originated (connection) errors are not distinguished from externally generated (transaction) errors and all end up 
+in the same bucket and are compared against 
+:ref:`outlier_detection.consecutive_5xx<envoy_api_field_cluster.OutlierDetection.consecutive_5xx>`,
+:ref:`outlier_detection.consecutive_gateway_failure<envoy_api_field_cluster.OutlierDetection.consecutive_gateway_failure>` and
+:ref:`outlier_detection.success_rate_stdev_factor<envoy_api_field_cluster.OutlierDetection.success_rate_stdev_factor>` 
+configuration items. For example, if connection to an upstream HTTP server fails twice because of timeout and 
+then, after successful connection, the server returns error code 500, the total error count will be 3. 
+
+Outlier detection may also be configured to distinguish locally originated (connection) errors from externally originated (transaction) errors. 
```

Comment:
All locally originated errors are not connection errors, right? Meaning, if a request times out it will fail in a locally originated sense, but it's not a connection level error? Is that right? If so can you clarify here and elsewhere?

Reply:
Did my best to explain locally originated errors.